cmd/oidc-broker/daemon/daemon_test.go

```diff
@@ -110,11 +110,53 @@ func TestAppCanQuitWithoutExecute(t *testing.T) {
 
 func TestAppRunFailsOnComponentsCreationAndQuit(t *testing.T) {
 	t.Parallel()
-	// Trigger the error with a cache directory that cannot be created over an
-	// existing file
-
-	// TODO
-	t.Fail()
+	const (
+		// Cache errors
+		dirIsFile = iota
+		wrongPermission
+		noParentDir
+	)
+
+	tests := map[string]struct {
+		cachePathBehavior int
+		configBehavior    int
+	}{
+		"Error on existing cache path being a file":    {cachePathBehavior: dirIsFile},
+		"Error on cache path missing parent directory": {cachePathBehavior: noParentDir},
+		"Error on wrong permission on cache path":      {cachePathBehavior: wrongPermission},
+	}
+	for name, tc := range tests {
+		tc := tc
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+
+			tmpDir := t.TempDir()
+			cachePath := filepath.Join(tmpDir, "cache")
+
+			switch tc.cachePathBehavior {
+			case dirIsFile:
+				err := os.WriteFile(cachePath, []byte("file"), 0600)
+				require.NoError(t, err, "Setup: could not create cache file for tests")
+			case wrongPermission:
+				err := os.Mkdir(cachePath, 0600)
+				require.NoError(t, err, "Setup: could not create cache directory for tests")
+			case noParentDir:
+				cachePath = filepath.Join(tmpDir, "doesnotexist", "cache")
+			}
+
+			config := daemon.DaemonConfig{
+				Verbosity: 0,
+				Paths: daemon.SystemPaths{
+					Cache: cachePath,
+				},
+			}
+
+			a := daemon.NewForTests(t, &config)
+			err := a.Run()
+			require.Error(t, err, "Run should return an error")
+			a.Quit()
```

Comment:
Honestly, I think we shouldn't even call `a.Quit()` now that I'm looking into it :joy: (I copied it from the `authd` daemon tests), since the daemon won't run, there's no point in calling quit (at least without asserting on anything)